virtool/auth/utils.py

```diff
@@ -7,33 +8,53 @@
     WriteAuthorizationModelRequest,
     TypeDefinition,
     Userset,
+    ApiClient,
 )
 from openfga_sdk.api import open_fga_api
 
-logger = getLogger("OpenFGA")
+logger = getLogger("openfga")
 
 
-async def connect_openfga():
+async def connect_openfga(fga_api_scheme: str, fga_api_host: str):
+    """
+    Connects to an OpenFGA server and configures the store id.
+    Returns the application client object.
+
+    :return: the client object
+
+    """
     configuration = openfga_sdk.Configuration(
-        api_scheme="http", api_host="localhost:8080"
+        api_scheme=fga_api_scheme, api_host=fga_api_host
     )
 
     logger.info("Connecting to OpenFGA")
 
     try:
         api_client = openfga_sdk.ApiClient(configuration)
+
+        await check_openfga_version(api_client)
+
         api_instance = open_fga_api.OpenFgaApi(api_client)
 
-        await create_store(api_instance, configuration)
+        store_id = await get_or_create_store(api_instance)
+
+        configuration.store_id = store_id
+
         await write_auth_model(api_instance)
 
     except ClientConnectorError:
         logger.fatal("Could not connect")
+        sys.exit(1)
 
     return api_client
 
 
-async def create_store(api_instance, configuration):
+async def get_or_create_store(api_instance):
```

Comment:
You can clean this function up quite a bit. Start by flipping the conditional so it is `if response.stores`. If you have `if not something: ... else` it should always be reversed to reduce mental overhead.

Then, think about how can further slim down the conditional.